tests/integ/modin/test_modin_stored_procedures.py

```diff
@@ -3,27 +3,51 @@
 # Copyright (c) 2012-2024 Snowflake Computing Inc. All rights reserved.
 #
 
+import os
+
 import modin.pandas as pd
+import pytest
 
 from snowflake.snowpark import Session
 from snowflake.snowpark.functions import sproc
 from snowflake.snowpark.modin.plugin import (
+    actual_pandas_version,
     supported_modin_version,
-    supported_pandas_version,
 )
 from tests.integ.utils.sql_counter import sql_count_checker
 
 PACKAGE_LIST = [
-    f"pandas=={supported_pandas_version}",
+    # modin 0.30.1 supports any pandas 2.2.x, so just pick whichever one is installed in the client
+    f"pandas=={actual_pandas_version}",
     f"modin=={supported_modin_version}",
     "snowflake-snowpark-python",
     "numpy",
 ]
 
+# Snowpark pandas strictly pins the modin dependency version, so while testing a dependency upgrade,
+# we need to upload snowflake-snowpark-python as a zip file. Otherwise, the conda package solver
+# will resolve snowflake-snowpark-python==1.16.0, the newest version which does not pin a modin
+# version.
+# We still specify snowflake-snowpark-python in the package list to prevent the sproc registration
+# code from failing in the solver step; the import here will override whatever version is chosen.
+IMPORT_LIST = [
```

Comment:
Did we do this because we want the current tests to pass? Is this supposedly the way our customer to use sproc?

Reply:
Customers should not need to do this, but this is necessary for tests to pass currently. As the comment says, even though `modin==0.30.1` is enabled for the testing account, since the latest version of `snowflake-snowpark-python` in anaconda pins `modin==0.28.1`, if we try to specify different modin version, the solver will resolve to an older version of the package that doesn't pin this dependency.

As long as the modin version is strictly pinned with `==` rather than `>=`, this will continue to be a problem moving forward whenever we try to upgrade the modin version.

Reply:
Discussed on Slack--I reverted this to just use latest `snowflake-snowpark-python`, and pin `pandas==2.2.1` and `modin==0.28.1` in this test (tracked by SNOW-1758760).